src/bytes/bytes.go

```diff
@@ -52,47 +86,162 @@ func explode(s []byte, n int) [][]byte {
+		//gobra:end-old-code 66239cff85d3bf25c4daa1d173a4c87733b57ef726a40dc7d98f63cf47ccdf38
+		res /* @ , indices @ */ = bytealg.Count(s,
+			/* @ unfolding acc(sl.Bytes(sep, 0, len(sep)), R40) in @ */ sep[0])
+		// @ assert forall i int :: { sl.View(s)[i:i+len(sep)] } 0 <= i && i < len(s) ==> sl.View(s)[i:i+len(sep)] == seq[byte]{sl.View(s)[i]}
+		return res // @ , indices
+		//gobra:endrewrite 66239cff85d3bf25c4daa1d173a4c87733b57ef726a40dc7d98f63cf47ccdf38
 	}
 	n := 0
+	// @ ghost olds := s
+	// @ ghost idx := 0
+	// @ ghost indices = set[int]{}
+	// @ invariant InRangeInc(idx, 0, len(olds))
+	// @ invariant olds[idx:] === s
+	// @ invariant n >= 0
+	// @ invariant len(indices) == n
+	// @ invariant forall j int :: {j in indices} j in indices ==> j < idx
+	// @ invariant acc(sl.Bytes(olds, 0, len(olds)), R39)
+	// @ invariant acc(sl.Bytes(sep, 0, len(sep)), R39)
+	// @ invariant forall j int :: { j in indices }{ sl.View(olds)[j:j+len(sep)] } j in indices ==> sl.View(olds)[j:j+len(sep)] == sl.View(sep) && !SetContainsInRange(indices, j-len(sep), j)
+	// @ invariant forall j int :: { j in indices }{ sl.View(olds)[j:j+len(sep)] } !(j in indices) ==> !(InRangeInc(j, 0, idx - len(sep))) || sl.View(olds)[j:j+len(sep)] != sl.View(sep) || SetContainsInRange(indices, j-len(sep), j)
+	// @ invariant forall j int :: { j in indices } j in indices ==> InRange(j, 0, idx-len(sep))
+	// @ invariant forall j int :: { j in indices } j in indices ==> InRange(j, 0, len(olds) - len(sep))
+	// @ invariant forall i, j int  :: { i in indices, j in indices }  (i in indices) && (j in indices) && i != j ==> (i + len(sep) < j) || (j + len(sep) < i)
+	// @ decreases len(s)
 	for {
+		// @ ghost p := R39 - R40 / 2
+		// @ unfold acc(sl.Bytes(olds, 0, len(olds)), p)
+
+		// @ assert forall j int :: {&olds[idx:][j]} 0 <= j && j < len(olds[idx:]) ==> &olds[idx:][j] == &olds[j+idx]
+
+		// @ fold acc(sl.Bytes(olds, 0, len(olds)), p)
+		// @ unfold acc(sl.Bytes(olds, 0, len(olds)), p)
+
+		// @ fold acc(sl.Bytes(s, 0, len(s)), p)
 		i := Index(s, sep)
+		// @ ghost vs := sl.View(s)
+		// @ unfold acc(sl.Bytes(s, 0, len(s)), p)
+
+		// @ fold acc(sl.Bytes(olds, 0, len(olds)), p)
+		// @ assert vs == sl.View(olds)[idx:]
 		if i == -1 {
-			return n
+
+			// @ ghost vs := sl.View(olds)
+			// @ ghost vsep := sl.View(sep)
+
+			// @ lemmaCountAux(sl.View(olds), sl.View(sep), indices, idx)
+
+			return n // @ , indices
 		}
+
+		// @ unfold acc(sl.Bytes(olds, 0, len(olds)), R39)
+
+		// @ unfold acc(sl.Bytes(sep, 0, len(sep)), R39)
+
+		// @ fold acc(sl.Bytes(sep, 0, len(sep)), R39)
+		// @ fold acc(sl.Bytes(olds, 0, len(olds)), R39)
+
+		// @ unfold acc(sl.Bytes(olds, 0, len(olds)), R39)
+		// @ unfold acc(sl.Bytes(sep, 0, len(sep)), R39)
+
+		// @ indices = indices union set[int]{idx+i}
+		// TODO: I dont understand why this doesn't hold
+		// @ assume forall i, j int  :: { i in indices, j in indices }  (i in indices) && (j in indices) ==> (i + len(sep) < j) || (j + len(sep) < i)
```

Comment:
isn't this assumption false? what if `i` and `j` are equal? then, your assumption is really implying a contradiciton. You can check whether that is the case with an `assert false` after the assumption